processing/src/main/java/org/apache/carbondata/processing/datatypes/PrimitiveDataType.java

```diff
@@ -240,141 +238,137 @@ public boolean getIsColumnDictionary() {
 
   @Override
   public void writeByteArray(Object input, DataOutputStream dataOutputStream,
```

Comment:
Can you break this function into smaller functions to make it more easy to understand

Reply:
This part of code is messed up. Need to check and refactor `useActualData` first (which needs some analysis). Without that these flows will be like this